### ISSUE TITLE
Column selection is reset when selecting with arrow keys

### ISSUE THREAD

**Issue by yurymann** [microsoft/vscode#205887]
Type: <b>Bug</b>

Steps to reproduce:

1. Enable column selection mode. 
2. Put cursor to beginning of a line and select several lines. 
3. Press `Ctrl+Shift+Right` several times to select several words in each line. 

   ![image](https://github.com/microsoft/vscode/assets/6237024/45a4ad78-61a2-4faf-a7ff-0bfd31417955)

4. Press `Shift+Left` to unselect the character to the left of the cursor in each line. 
    -  Expected: one character is unselected in each line.
       
       ![image](https://github.com/microsoft/vscode/assets/6237024/17abb1c5-1b3a-4dcc-a9e0-85545fdcd8ca)

    - Actual: the column selection is lost and the character is unselected only in the first line. 

      ![image](https://github.com/microsoft/vscode/assets/6237024/03e99a50-63ac-4b30-91da-65e7c092f9b4)

`Shift+Right` in Step 4 breaks it the same way. 

If you start the column selection with `Shift+Right` rather than `Ctrl+Shift+Right`, then everything works. It looks like the combinatoin of word selection with subsequent character selection breaks it. 

VS Code version: Code 1.86.2 (903b1e9d8990623e3d7da1df3d33db3e42d80eda, 2024-02-13T19:40:56.878Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.82GB (5.32GB free)|
|Process Argv|--crash-reporter-id 78edb399-aebd-4224-8197-f1211e83f080|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
scratchpads|bue|1.0.0
gitlens|eam|2024.2.2104
intellij-idea-keybindings|k--|1.5.13
debugpy|ms-|2024.0.0
python|ms-|2024.0.1
vscode-pylance|ms-|2024.2.2
remote-wsl|ms-|0.86.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
vscrpc:30673769
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
pydisww2:30959801
edj9j583:30969295
a89i1917:30961429
7j2b6412:30969109
bg6jg535:30958357

```

</details>

**Comment by IllusionMH** [comment]
/needsMoreInfo

Is this problem reproducible in latest VS Code (1.92)?

**Comment by yurymann** [comment]
@IllusionMH, yes, it still exists in `1.92.0`. It's very easy to reproduce and probably easy to fix. Thanks in advance!
